cmd/jaeger/internal/extension/remotesampling/internal/http.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+import (
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/pkg/clientcfg/clientcfghttp"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"github.com/jaegertracing/jaeger/pkg/recoveryhandler"
+)
+
+func StartHttpServer(ss strategystore.StrategyStore, logger *zap.Logger, port string) (*http.Server, error) {
```

Comment:
Are you sure OTEL Collector does not already have a helper function for this? They do start various HTTP servers. I would check OTLP/HTTP receiver, z-pages extension.

Reply:
I am not able to find any helper function. 

In both OTLP/HTTP receiver, z-pages extension, http is initiated using `http.NewServeMux()`.